### PR TITLE
Feature/file manager

### LINKED ISSUE #21: Fix file manager
Change the file manager module to use local file system for storage instead of Amazon S3

### PR DESCRIPTION
**Pull request by aabounegm**
Resolves #21

### CODE REVIEW

File: innopoints/core/file_manager/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"""File manager module."""
+import os
+from .s3 import FileManagerS3
+from .local import FileManagerLocal
+
```
Comment (illright) [inline innopoints/core/file_manager/__init__.py:6]: Two blank lines? :)

File: innopoints/core/file_manager/__init__.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"""File manager module."""
+import os
+from .s3 import FileManagerS3
```
Comment (illright) [inline innopoints/core/file_manager/__init__.py:4]: Also would be nice to have a line between native and local imports

File: innopoints/core/file_manager/base.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+"""Provides a base interface for the file management modules to implement"""
+from abc import ABC, abstractmethod
+from werkzeug.datastructures import FileStorage
+
+
+class FileManagerBase(ABC):
+    """Base abstract class as an interface for file managers"""
+    @abstractmethod
+    def __init__(self, url = ''):
```
Comment (illright) [inline innopoints/core/file_manager/base.py]: Spaces around the `=` in parameters, ouch

File: innopoints/core/file_manager/base.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        super().__init__()
+        self.BASE_PATH = url
+
+    @abstractmethod
+    def retrieve(self, handle: str, namespace: str) -> bytes:
+        """Get the file with the given handle and namespace"""
```
Comment (illright) [inline innopoints/core/file_manager/base.py]: Also give all docstrings a period at the end. Let's stick to this style

File: innopoints/core/file_manager/local.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+from .base import FileManagerBase
+
+
+class FileManagerLocal(FileManagerBase):
+    """Implementation of the file manager using local file system"""
+    def __init__(self, url='./static_files'):
+        super().__init__(url)
+        if not os.path.exists(url):
+            os.makedirs(url)
+
```
Comment (illright) [inline innopoints/core/file_manager/local.py:15]: Single line between methods

File: innopoints/core/file_manager/local.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    def delete(self, handle: str, namespace: str):
+        """Delete the file with a given handle from the namespace"""
+        folder = self._join_base(namespace)
+        if not os.path.exists(folder):
+            raise FileNotFoundError()
+        filename = os.path.join(folder, handle)
+        os.remove(filename)
+        # directory is now empty ?
+        if not os.listdir(folder):
```
Comment (illright) [inline innopoints/core/file_manager/local.py:44]: I wouldn't bother

File: innopoints/core/file_manager/s3.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
+"""Manages static files. This particular module uses the AWS S3 to store files"""
+
+import requests
+from .base import FileManagerBase, FileStorage
+
```
Comment (illright) [inline innopoints/core/file_manager/s3.py:6]: Lines...

**Review by illright — CHANGES_REQUESTED** [review]

File: innopoints/core/file_manager/base.py (hunk 3 above)
Comment (aabounegm) [reply innopoints/core/file_manager/base.py]: ooh. I guess it's the habit of always surrounding `=` with spaces. I didn't notice 😅

File: innopoints/core/file_manager/local.py (hunk 6 above)
Comment (aabounegm) [reply innopoints/core/file_manager/local.py:44]: I already bothered))

**Pull request merged**